### PR TITLE
Fs 02 auth

### LINKED ISSUE #3: Feat: "Auth"

### PR DESCRIPTION
**Pull request by sadathshariff**
Implemented 
1. Auth
2. Private Routes
3. Login and Signup
4. used useLocation Hook

### CODE REVIEW

File: src/components/RequireAuth/RequireAuth.jsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { useAuth } from "../../context";
+import { Navigate, useLocation } from "react-router-dom";
+export const RequireAuth = ({ children }) => {
+  const { authState } = useAuth();
+  const { isLoggedIn } = authState;
+  const location = useLocation();
+  return isLoggedIn ? (
+    children
+  ) : (
+    <Navigate to="/login" state={{ from: location }} replace />
```
Comment (nishasen) [inline src/components/RequireAuth/RequireAuth.jsx:10]: like how you have used useLocation

File: src/components/Sidenav/Sidenav.css [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+@media screen and (max-width: 960px) {
+  .sidenav-container {
+    display: none;
+  }
+
+  .sidenav-container-display {
+    display: block;
+    position: fixed;
+    top: 85px;
+    /* bottom: 0;
```
Comment (nishasen) [inline src/components/Sidenav/Sidenav.css]: remove the commented code

File: src/components/Sidenav/Sidenav.css [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    left: 0; */
```
Comment (aishwarya-bm) [inline src/components/Sidenav/Sidenav.css]: remove comments

File: src/components/Sidenav/Sidenav.css [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    background: var(--text-color);
+    color: var(--off-white);
+    z-index: 3;
+    transition: all 1s ease;
+    flex-basis: 20%;
+    padding: 1rem;
+    height: 100vh;
+    border-radius: 5px;
+    box-shadow: 0 6px 6px 2px rgba(0, 0, 0, 0.5);
```
Comment (aishwarya-bm) [inline src/components/Sidenav/Sidenav.css:33]: give color variable

File: src/context/Auth/util.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    } else {
+      ToastMsg("Something went wrong,Try again later!", "error");
+    }
+  }
+};
+
+export const handleSingup = async (signupDetails, authDispatch, navigate) => {
+  try {
+    const res = await axios.post("/api/auth/signup", signupDetails);
+    console.log(res);
```
Comment (aishwarya-bm) [inline src/context/Auth/util.js]: remove console log

**Review by nishasen — APPROVED** [review]
Liked how you have organized everything so well and optimised small small parts, learnt a lot

**Review by aishwarya-bm — APPROVED** [review]
Auth implementation and directing to the intended page after login functionality is awesome! Great work! Left very minor suggestions.

File: src/components/Sidenav/Sidenav.css (hunk 2 above)
Comment (sadathshariff) [reply src/components/Sidenav/Sidenav.css]: Thankyou for pointing out

File: src/context/Auth/util.js (hunk 5 above)
Comment (sadathshariff) [reply src/context/Auth/util.js]: Thankyou for pointing out

**Pull request merged**